src/main/java/echobot/Command.java

```diff
@@ -0,0 +1,367 @@
+package echobot;
+import echobot.task.*;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+/**
+ * Represents a command issued by the user.
+ * The command includes the action to be performed and the associated task list.
+ */
+public class Command {
+    private final String[] inputParts;
+    private final String command;
+    private final TaskList tasks;
+
+    /**
+     * Constructs a Command with the specified input parts and task list.
+     *
+     * @param inputParts The parts of the user input.
+     * @param tasks The list of tasks to be manipulated by this command.
+     */
+    public Command(String[] inputParts, TaskList tasks) {
+        this.inputParts = inputParts;
+        this.command = inputParts[0];
+        this.tasks = tasks;
+    }
+
+    /**
+     * Returns the command string.
+     *
+     * @return The command string.
+     */
+    public String getCommand() {
+        return this.command;
+    }
+
+    /**
+     * Checks if the command is an exit command.
+     *
+     * @return True if the command is "bye", false otherwise.
+     */
+    public boolean isExit() {
+        return this.command.equals("bye");
+    }
+
+    /**
+     * Executes the command based on its type.
+     * The command may modify the task list or provide feedback to the user.
+     */
+    public void run() {
+        switch (this.command) {
+            case "bye":
```

Comment:
Kindly indent your switch code. For instance, the 'case' keyword should be indented towards the left inside the switch block of code.